### DISCUSSION TITLE
Maintain aspect ratio when zooming window

### DISCUSSION THREAD

**Discussion by brentonhershner** [rxhanson/RectanglePro-Community#453] [Ideas]
It would be nice if windows approximately maintained their aspect ratio when zooming with the keyboard shortcut.
Especially when I snap a window to a corner and then enlarge it, it just feels like it should grow towards the opposite corner and eventually fill the whole window, rather than first filling vertically before filling the window out horizontally (on a monitor in landscape).

**Comment by rxhanson** [comment]
If I understand correctly, I think what you're looking for is achieved by executing this terminal command, then restarting the app:

```
defaults write com.knollsoft.Hookshot curtainChangeSize -int 2
```

**Reply by brentonhershner** [reply]
Thank you for the reply. These hidden preferences are awesome!
But changing curtainChangeSize didn't change the behavior in the way I was hoping for. For me, this only changed the way the placement of the window when making the window smaller.

If I grow a window, it will grow equally in height and width. For example, if I start with a 16:9 aspect ratio (example: 800 x 450), if it grows by 600 pixels, it will become more square at 4:3 (1400 x 1050).
It would be really nice if it could grow proportionally and maintain the aspect ratio. For example, starting with 16:9 (800 x 450), if it grows by 600 pixels in width, it proportionally only grows in 338 pixels, resulting in 1400 x 788, maintaining the 16:9 aspect ratio.

I'm not expecting this, but I just thought it would be a nice enhancement for you to consider. Thank you for your time and great app!

**Reply by rxhanson** [reply]
Thanks for the detailed reply. I understand what you're looking for now. At a time when I'm back in that part of the code, I'll see what I can do.

**Reply by brentonhershner** [reply]
Thank you!
